energy-integration/timestamp-oracle/src/epoch_to_timestamp.rs

```diff
@@ -0,0 +1,69 @@
+use common_structs::{Epoch, Timestamp};
+
+multiversx_sc::imports!();
+
+#[multiversx_sc::module]
+pub trait EpochToTimestampModule {
+    #[only_owner]
+    #[endpoint(setStartTimestampForEpoch)]
+    fn set_start_timestamp_for_epoch(&self, epoch: Epoch, start_timestamp: Timestamp) {
+        let mapper = self.timestamp_for_epoch(epoch);
+        require!(
+            mapper.is_empty(),
+            "Overwriting timestamp. If you're sure about this, use the overwriteStartTimestampForEpoch endpoint"
+        );
+
+        mapper.set(start_timestamp);
+    }
+
+    #[only_owner]
+    #[endpoint(overwriteStartTimestampForEpoch)]
+    fn overwrite_start_timestamp_for_epoch(&self, epoch: Epoch, start_timestamp: Timestamp) {
+        self.timestamp_for_epoch(epoch).set(start_timestamp);
+    }
+
+    #[endpoint(updateAndGetTimestampStartEpoch)]
+    fn update_and_get_timestamp_start_epoch(&self) -> Timestamp {
+        let current_epoch = self.blockchain().get_block_epoch();
+        let last_update_epoch = self.epoch_last_interaction().get();
+        let mapper = self.timestamp_for_epoch(current_epoch);
+        if current_epoch == last_update_epoch {
+            return mapper.get();
```

Comment:
Here we should always have a value. But because we have external factors, like the owner could update the storage with overwrite, I would also put a check here that the value stored by timestamp_for_epoch is not 0. And I would go even to a require, to completely crash the execution in that particular case, to not negatively impact the rewards computation.

Reply:
I like the new implementation, where you don't need to terminate the execution, but return clearly that there is no data.